### PR TITLE
Update http request mapping

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Updating as per common schema mapping rule

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryPartBTests.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            var httpUrl = "https://www.foo.bar/search";
+            var httpResponsCode = "0";
+            activity.SetStatus(Status.Ok);
+            activity.SetTag(SemanticConventions.AttributeHttpUrl, httpUrl);
+            activity.SetTag(SemanticConventions.AttributeHttpStatusCode, null);
+            var requestData = TelemetryPartB.GetRequestData(activity);
+            Assert.Equal(requestData.Name, activity.DisplayName);
+            Assert.Equal(requestData.Id, activity.Context.SpanId.ToHexString());
+            Assert.Equal(requestData.Url, httpUrl);
+            Assert.Equal(requestData.ResponseCode, httpResponsCode);
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryPartBTests.cs:54]: nit:
1. convention - put the expected result as the 1st argument `Assert.Equal(expectedValue, actualValue)`.
2. consider removing `var httpResponseCode = "0"` and directly put the literal constant inline.

**Review by reyang — APPROVED** [review]
LGTM.

**Review by rajkumar-rangaraj — APPROVED** [review]

**Comment by vishweshbankwar** [comment]
Need to update this PR to get url from other attributes if `http.url` is not present. Converting it to draft for now.

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryPartBTests.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        public void ValidateRequestDataSuccess(string activityStatus)
+        {
+            using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+            using var activity = activitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Server,
+                parentContext: new ActivityContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), ActivityTraceFlags.Recorded),
+                startTime: DateTime.UtcNow);
+
+            if (activityStatus == "Ok")
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryPartBTests.cs:96]: `Status` could be used as test args. In that way we could avoid condition check here.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/HttpHelperTests.cs [hunk 3]
```diff
@@ -183,6 +183,131 @@ public void GetUrl_HttpUrl_Success()
...
+
+            string expectedUrl = "http://www.httphost.org/path";
+            string url = PartBTags.GetRequestUrl();
+            Assert.Equal(expectedUrl, url);
+        }
+
+        [Theory]
+        [InlineData("80")]
+        [InlineData("443")]
+        [InlineData("8888")]
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/HttpHelperTests.cs:212]: Looks like port is a `long` type in OpenTelemetry SDK. Please validate, not a blocker for this PR.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/HttpHelper.cs [hunk 4]
```diff
@@ -92,5 +92,70 @@ internal static void GenerateUrlAndAuthority(this AzMonList tagObjects, out stri
...
+                if (!string.IsNullOrWhiteSpace(host))
+                {
+                    var netHostPort = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeNetHostPort)?.ToString();
+                    if (!string.IsNullOrWhiteSpace(netHostPort))
+                    {
+                        string colon = ":";
+                        if (netHostPort == "80" || netHostPort == "443")
+                        {
+                            netHostPort = "";
+                            colon = "";
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/HttpHelper.cs:150]: This (assigning empty string to `colon`) might be counter intuitive, would this be better?
```csharp
if (netHostPort == "80" || netHostPort == "443")
{
    return $"{httpScheme}://{host}{httpTarget}";
}
return $"{httpScheme}://{host}:{netHostPort}{httpTarget}";
```

**Review by reyang — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryPartBTests.cs (hunk 2 above)
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryPartBTests.cs:96]: it does not allow me to pass Status.Ok etc. as args. Gives this [error](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0182?f1url=%3FappId%3Droslyn%26k%3Dk(CS0182))

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/HttpHelperTests.cs (hunk 3 above)
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/HttpHelperTests.cs:212]: Defined as int in [spec](: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#general-network-connection-attributes)
Will add this to do list and update all tets

**Pull request merged**